### ISSUE TITLE
Unexpected trimming of whitespace after loop

### ISSUE THREAD

**Issue by mlafeldt** [sunng87/handlebars-rust#472]
Given a template like this:

```hbs
{{#each routes}}
import { default as {{this.handler}} } from '{{this.file_path}}'
{{/each}}

addEventListener('fetch', (event) => {
  event.respondWith(handleEvent(event))
})
```

handlebars 4.1.3 will remove the empty line after the loop (in fact, it will remove all empty lines if there are any):

```js
import { default as __hello_handler } from './hello.js'
import { default as __world_index_handler } from './world/index.js'
import { default as __index_handler } from './index.js'
addEventListener('fetch', (event) => {
  event.respondWith(handleEvent(event))
})
```

I'm wondering why that happens though? I don't use `~` or anything else that controls whitespace. My understanding is that rendering should not change any non-templating bits.

**Comment by mlafeldt** [comment]
Might be related to #468

**Comment by sunng87** [comment]
I can reproduce this issue. Thanks for reporting. 

It has something to do with our empty line trimming strategy. We should trim at most one line. I will create a fix in days.

**Comment by sunng87** [comment]
Released in 4.1.4
